projects/demo-playwright/tests/kit/input-phone/input-phone.spec.ts

```diff
@@ -80,4 +86,944 @@ test.describe('InputPhone', () => {
+                    await expect(example).toHaveScreenshot('14-input-phone.png');
+                });
+            });
+
+            test.describe('Text selection', () => {
+                // ab|cd|e – it means that in string "abcde" characters "cd" are selected
+
+                test.beforeEach(async ({page}) => {
+                    await input.fill('9123456789');
+                    await page.waitForTimeout(100);
+                });
+
+                test.describe('Select range and press "Backspace"', () => {
+                    test('+7 (912) 345-67-|89| => Backspace => +7 (912) 345-67|', async ({
+                        page,
+                    }) => {
+                        await page.keyboard.down('Shift');
+
+                        for (let i = 0; i < 2; i++) {
+                            await page.keyboard.press('ArrowLeft');
+                        }
+
+                        await page.keyboard.up('Shift');
+                        await page.keyboard.press('Backspace');
+                        await page.waitForTimeout(100);
+                        await expect(input).toHaveJSProperty('selectionStart', 15);
+                        await expect(input).toHaveJSProperty('selectionEnd', 15);
+                        await expect(example).toHaveScreenshot('15-input-phone.png');
+                    });
+
+                    test('+7 (912) 345-6|7-89| => Backspace => +7 (912) 345-6|', async ({
+                        page,
+                    }) => {
+                        await page.keyboard.down('Shift');
+
+                        for (let i = 0; i < 4; i++) {
+                            await page.keyboard.press('ArrowLeft');
+                        }
+
+                        await page.keyboard.up('Shift');
+                        await page.keyboard.press('Backspace');
+                        await page.waitForTimeout(100);
+                        await expect(input).toHaveJSProperty('selectionStart', 14);
+                        await expect(input).toHaveJSProperty('selectionEnd', 14);
+                        await expect(example).toHaveScreenshot('16-input-phone.png');
+                    });
+
+                    test('+7 (912) 345-6|7-8|9 => Backspace => +7 (912) 345-6|9', async ({
+                        page,
+                    }) => {
+                        await page.keyboard.press('ArrowLeft');
+                        await page.keyboard.down('Shift');
+
+                        for (let i = 0; i < 3; i++) {
+                            await page.keyboard.press('ArrowLeft');
+                        }
+
+                        await page.keyboard.up('Shift');
+                        await page.keyboard.press('Backspace');
+                        await page.waitForTimeout(100);
+                        await expect(input).toHaveJSProperty('selectionStart', 14);
+                        await expect(input).toHaveJSProperty('selectionEnd', 14);
```

Comment:
It is difficult to read the such assertion.
Let's use this pattern:

```suggestion
                        await expect(input).toHaveJSProperty('selectionStart', '+7 (912) 345-6'.length);
                        await expect(input).toHaveJSProperty('selectionEnd', '+7 (912) 345-6'.length);
```